nixos/modules/services/misc/ethminer.nix

```diff
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ethminer;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.ethminer = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable ethminer ether mining.";
+      };
+
+      recheckInterval = mkOption {
+        type = types.int;
+        default = 2000;
+        description = "Interval in milliseconds to farm recheck.";
+      };
+
+      toolkit = mkOption {
+        type = types.enum [ "cuda" "opencl" ];
+        default = "cuda";
+        description = "Cuda or opencl toolkit.";
+      };
+
+      apiPort = mkOption {
+        type = types.int;
+        default = -3333;
+        description = "Ethminer api port. minus sign puts api in read-only mode.";
+      };
+
+      wallet = mkOption {
+        type = types.str;
+        example = "1x0484A1EfB3C88B0C7dE21919D480fF51099c8113";
```

Comment:
You still haven't changed this! This is getting annoying, I can't trust you like this if you mark everything as resolved when in fact it's still the same. Yes you changed the "0x" to "1x", but "0x" is supposed to signify that it's a hex address, "1x" is invalid for that

Reply:
> a random (invalid) one would be better.
by doing `1x` it becomes invalid.  im not sure how else to ensure an address is invalid?  i will increment the last char, instead of the first, as i think that is what you are suggesting.

Reply:
Well of course you can't guarantee it, but if you generate a random number, the chance of it corresponding to a valid address is in the order of 0.0000000001% or so. We still want plausible examples though, which an address starting with `1x` is not. `0x01234...` is perfect though, it even signifies that this is an example value, and it's just as improbably to be an actual address as a random number.

Reply:
updated with the more example-like example (`0x0123456789abcdef0123456789abcdef01234567`)